src/firmware_new/boards/frankie_v1/io/allegro_a3931_motor_driver.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+/**
+ * This file is an abstraction around the Allegro A3931 Motor Driver
+ */
+
+#include "firmware_new/boards/frankie_v1/Drivers/STM32H7xx_HAL_Driver/Inc/stm32h7xx_hal.h"
+#include "firmware_new/boards/frankie_v1/io/gpio_pin.h"
+#include "firmware_new/boards/frankie_v1/io/pwm_pin.h"
+
+typedef enum
+{
+    CLOCKWISE,
+    COUNTERCLOCKWISE
+} AllegroA3931MotorDriverDriveDirection;
+
+typedef struct AllegroA3931MotorDriver AllegroA3931MotorDriver_t;
+
+/**
+ * Create a Motor Driver
+ * @param pwm_pin The PWM pin to the driver. Note that this must be configured to
+ *                operate in the correct frequency range for the motor driver. Please
+ *                talk to Elec. to determine this.
+ * @param reset_pin A GPIO pin connected to the "reset" pin on the motor driver.
+ * @param coast_pin A GPIO pin connected to the "coast" pin on the motor driver.
+ * @param mode_pin A GPIO pin connected to the "mode" pin on the motor driver.
+ * @param direction_pin A GPIO pin connected to the "direction" pin on the motor driver.
+ * @param brake_pin A GPIO pin connected to the "brake" pin on the motor driver.
+ * @param esf_pin A GPIO pin connected to the "ESF" pin on the motor driver.
+ *
+ * @return A motor driver with the pwm percentage set to zero.
+ */
+AllegroA3931MotorDriver_t* io_allegro_a3931_motor_driver_create(
+    PwmPin_t* pwm_pin, GpioPin_t* reset_pin, GpioPin_t* coast_pin, GpioPin_t* mode_pin,
+    GpioPin_t* direction_pin, GpioPin_t* brake_pin, GpioPin_t* esf_pin);
+
+/**
+ * Destroy the given motor driver
+ * @param motor_driver The motor driver to destroy
+ */
+void io_allegro_a3931_motor_driver_destroy(AllegroA3931MotorDriver_t* motor_driver);
+
+/**
+ * Set the rotation direction for the given motor driver
+ *
+ * Note that the rotation is from the perspective of rear of the motor, looking out
+ * along the motor shaft
+ *
+ * @param motor_driver
+ * @param direction
+ */
+void io_allegro_a3931_motor_driver_setDirection(
+    AllegroA3931MotorDriver_t* motor_driver,
+    AllegroA3931MotorDriverDriveDirection direction);
+
+/**
+ * Set the PWM percentage for the given motor driver
+ * @param motor_driver
+ * @param pwm_percentage A value in [0,1] indicating the PWM percentage
```

Comment:
IMO, 0-100 would be more intuitive.

Major nit though

Reply:
This is definitely convention, but I don't think I've ever seen percentage represented this way. Generally it's not convenient because if you want to scale something by a percentage you then need to divide by 100.

Reply:
Yea I think [0, 1] is slightly better here